### PR TITLE
Add First Class Closure rules

### PR DESCRIPTION
**Pull request by Crell**
Not much else to say here.  It's pretty basic, and that's a good thing.

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -1047,6 +1047,13 @@ $foo->bar(
 );
 ```
 
+### 7.1 Function callable references
+
+A function may be referenced in a way that creates a closure out of it, by providing `...` in place of arguments.
```
Comment (samdark) [inline spec.md:1052]: Need an example.

Comment (Crell) [reply spec.md:1052]: There literally is one in the very next sentence. :smile:

Comment (samdark) [reply spec.md:1052]: Seems I've missed a PHP feature :)

**Review by samdark — APPROVED** [review]

**Comment by KorvinSzanto** [comment]
This looks good to me, I can't help but feel like this belongs in the operators section but I don't really know that the `...` here would be an operator.

**Comment by Crell** [comment]
It's not really an operator.  It's really its own "thing".  If anything, it's an oddball function call syntax, so I suppose it could go there?  I put it in the closures section, but thinking about it, putting it in the function invocation section could make more sense.

**Comment by Crell** [comment]
Rebased, and moved to the function/method section.

**Comment by kenguest** [comment]
Just to tighten up the text, perhaps "The correct format is `foo(...)`." would suffice?

**Comment by Crell** [comment]
It feels like it flows better to me with the connecting clause, but if @KorvinSzanto would favor without it I won't take offense. :-)

**Comment by KorvinSzanto** [comment]
I prefer it without but I prefer to do the finer editing once the entire doc is closer to being ready so we can try to get a common voice.

**Pull request merged**